### PR TITLE
MM-37456 "Threads" item in the sidebar should not scroll

### PR DESCRIPTION
**Pull request by anurag6713**
#### Summary
Makes the "Threads" item in the sidebar as fixed.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-37456

#### Device Information
This PR was tested on: iOS 14.5, iPhone 12 Simulator, Android 11 Pixel XL emulator

#### Screenshots

https://user-images.githubusercontent.com/3680799/127208468-cff3d0e5-e99c-4da9-b27b-d7b0323d01d1.mp4

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: app/components/sidebars/main/channels_list/list/list.js [hunk 1]
```diff
@@ -414,16 +400,24 @@ export default class List extends PureComponent {
...
 
         return (
             <View
                 style={styles.container}
                 onLayout={this.onLayout}
             >
+                {collapsedThreadsEnabled && (
+                    <ThreadsSidebarEntry/>
+                )}
                 <SectionList
```
Comment (anurag6713) [inline app/components/sidebars/main/channels_list/list/list.js:421]: Tried using `stickyHeaderIndices` but it didn't work so ended up placing the component above the `SectionList`

File: app/components/sidebars/main/channels_list/list/index.js [hunk 2]
```diff
@@ -50,7 +50,6 @@ function mapStateToProps(state) {
         sidebarPrefs.sorting,
         true, // The mobile app should always display the Unreads section regardless of user settings (MM-13420)
         sidebarPrefs.favorite_at_top === 'true' && favoriteChannelIds.length,
-        collapsedThreadsEnabled,
```
Comment (enahum) [inline app/components/sidebars/main/channels_list/list/index.js:53]: ?

File: app/components/sidebars/main/threads_entry/threads_entry.tsx [hunk 3]
```diff
@@ -97,40 +97,46 @@ const ThreadsEntry = ({
...
+                        <Text
+                            style={textStyle}
+                            ellipsizeMode='tail'
+                            numberOfLines={1}
+                        >
+                            {intl.formatMessage({
+                                id: 'threads',
+                                defaultMessage: 'Threads',
+                            })}
+                        </Text>
```
Comment (enahum) [inline app/components/sidebars/main/threads_entry/threads_entry.tsx:123]: you can use FormattedText here

File: app/components/sidebars/main/channels_list/list/index.js (hunk 2 above)
Comment (anurag6713) [reply app/components/sidebars/main/channels_list/list/index.js:53]: We were sending this to add "Threads" to the sections. As we don't need it anymore, I have removed that code.

File: app/components/sidebars/main/threads_entry/threads_entry.tsx [hunk 4]
```diff
@@ -97,40 +97,46 @@ const ThreadsEntry = ({
...
 
 const getExtraStyleSheet = makeStyleFromTheme((theme: Theme) => {
     return {
-        container: {
+        baseContainer: {
             marginTop: 16,
+            marginBottom: 12,
+        },
+        container: {
+            flex: 0, // Override the existing flex: 1
```
Comment (enahum) [inline app/components/sidebars/main/threads_entry/threads_entry.tsx:139]: where is flex:1 defined? you may not even need this

**Review by enahum — APPROVED** [review]
@shazm FYI

**Comment by enahum** [comment]
@anurag6713 tests failed

File: app/components/sidebars/main/channels_list/list/index.js (hunk 2 above)
Comment (enahum) [reply app/components/sidebars/main/channels_list/list/index.js:53]: I see this prop being used in the list component, please double check

File: app/components/sidebars/main/channels_list/list/list.js [hunk 5]
```diff
@@ -414,16 +400,24 @@ export default class List extends PureComponent {
...
+        if (collapsedThreadsEnabled) {
+            indicatorStyle.push({marginTop: 70});
+        }
 
         return (
             <View
                 style={styles.container}
                 onLayout={this.onLayout}
             >
+                {collapsedThreadsEnabled && (
```
Comment (enahum) [inline app/components/sidebars/main/channels_list/list/list.js:418]: seems to be used here

File: app/components/sidebars/main/threads_entry/threads_entry.tsx (hunk 4 above)
Comment (anurag6713) [reply app/components/sidebars/main/threads_entry/threads_entry.tsx:139]: We are using the styles from `app/components/sidebars/main/channels_list/channel_item/channel_item.js` as we need visually the same styling as rest of the items and overriding it from here.

File: app/components/sidebars/main/channels_list/list/list.js (hunk 5 above)
Comment (anurag6713) [reply app/components/sidebars/main/channels_list/list/list.js:418]: Yes we are using it here but removed from the function we are passing to build the sections 🙂

Comment (enahum) [reply app/components/sidebars/main/channels_list/list/list.js:418]: Oh sorry the diff wasn't clearly showing that it was as an arg for the selector

File: app/components/sidebars/main/threads_entry/threads_entry.tsx (hunk 4 above)
Comment (enahum) [reply app/components/sidebars/main/threads_entry/threads_entry.tsx:132]: As in extraStyle? Cause extraStyle is overriding this flex prop if defined

Comment (anurag6713) [reply app/components/sidebars/main/threads_entry/threads_entry.tsx:132]: Yes .. this is the extraStyle. Actual style is the imported one.

Comment (enahum) [reply app/components/sidebars/main/threads_entry/threads_entry.tsx:132]: Oh I see... weird naming then.. not blocking though

**Comment by jgilliam17** [comment]
Thanks @anurag6713 Looks good, I'll let Matthew have a pass first before approving. 
One question - Threads item seems a bit wide, is this intended?
IMG_CA706B18029C-1

**Review by matthewbirtch — CHANGES_REQUESTED** [review]
@anurag6713 this looks good. I think we should reduce the space below threads here though. Can we remove at least 8px of space?

image

**Comment by anurag6713** [comment]
@matthewbirtch .. removed 8px. 
Untitled-1

**Comment by anurag6713** [comment]
> One question - Threads item seems a bit wide, is this intended?
Thank you @jgilliam17. Styling is same as the channels item in the list. We are reusing the same styles as the rest of them.

**Comment by matthewbirtch** [comment]
Looks good @anurag6713 based on the screenshots. Will do one last look at the new build and then I'll approve

**Review by matthewbirtch — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @anurag6713 
Tested, looks good to merge.
- Verified Threads item on the channel sidebar no longer scrolls.

**Pull request merged**